code/modules/power/engines/singularity/emitter.dm

```diff
@@ -302,6 +302,8 @@
 	else
 		fire_delay = rand(minimum_fire_delay, maximum_fire_delay)
 		shot_number = 0
+
+	P.firer = src
```

Comment:
How do you suggest we check that the firer was an emitter then?

Reply:
Could just check if it is a hitscan emitter, which is fired by emitters. Cyborg emitter projectiles are no longer hitscan. This would let you differentiate from standard emitter useage, and cyborgs

Reply:
I'd prefer if that wasn't the final solution but it does work

Reply:
Done